.github/workflows/dev-deploy.yml

```diff
@@ -7,36 +7,79 @@ on:
   delete:
     branches:
       - 'dev-*'
+  pull_request:
+    types: [labeled, unlabeled, synchronize, closed]
 
 # https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
 permissions:
-  id-token: write
   contents: read
+  id-token: write
+  pull-requests: write
 
 jobs:
   deploy-to-dev:
     name: deploy dev branch
     runs-on: ubuntu-latest
     environment: dev
-    if: github.repository == 'chanzuckerberg/cryoet-data-portal'
+    if: |
+      github.repository == 'chanzuckerberg/cryoet-data-portal' && (
+        github.event_name != 'pull_request' || (
+          startsWith(github.head_ref, 'dev-') != true && (
+            contains(github.event.pull_request.labels.*.name, 'preview') || (
+              github.event.action == 'unlabeled' &&
+              github.event.label.name == 'preview'
+            )
+          )
+        )
+      )
+
     steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - id: deploy-data
+        name: get stack name
+        uses: actions/github-script@v7
+        with:
+          result-encoding: string
+          script: |
+            const event = ${{ toJson(github.event) }}
+            const eventName = '${{ github.event_name }}'
+
+            let name = ''
+
+            switch (eventName) {
+              case 'pull_request': {
+                name = event.pull_request.head.ref
+                break
+              }
+
+              default: {
+                name = event.ref.replace('refs/heads/', '')
+                break
+              }
+            }
+
+            return name.replaceAll('/', '-').slice(0, 25)
+
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v4
         with:
           mask-aws-account-id: true
           aws-region: ${{ secrets.AWS_REGION }}
           role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
           role-duration-seconds: 1200
-      - name: Setting up stack name
-        run: |
-          if [[ ${#GITHUB_REF_NAME} -gt 25 ]]; then
-            echo "branch name: ${GITHUB_REF_NAME}, length ${#GITHUB_REF_NAME}, is $((${#GITHUB_REF_NAME} - 25)) characters too long, please use a branch name that's 25 characters or shorter"
-            exit 1
-          else
-            echo $GITHUB_REF_NAME | tr '[:upper:]' '[:lower:]' | xargs -I {} -n 1 echo STACK_NAME={} >> $GITHUB_ENV
-          fi
+
       - name: Create or update dev stack
-        if: ${{ github.event_name == 'push' }}
+        if: |
+          github.event_name == 'push' || (
+            github.event_name == 'pull_request' && (
+              github.event.action == 'labeled' || (
+                github.event.action == 'synchronize' &&
+                contains(github.event.pull_request.labels.*.name, 'preview')
```

Comment:
is this the only thing you need to check? this whole condition only has to be a subset of the condition at the job level right?